qiskit/optimization/problems/linear_expression.py

```diff
@@ -154,3 +154,17 @@ def evaluate(self, x: Union[ndarray, List, Dict[Union[int, str], float]]) -> flo
 
         # return the result
         return val
+
+    # pylint: disable=unused-argument
+    def evaluate_gradient(self, x: Union[ndarray, List, Dict[Union[int, str], float]]) -> ndarray:
```

Comment:
I see the same method signature in quadratic expression - there is no base class or anything defining this signature as common right? The place where its called (quadratic objective) knows the type of linear or quadratic. Maybe its convenient to keep the same signature on the linear and quadratic, even though its not a signature  from common base class. But it seems x is just not used for linear, so without a common base class I guess I am wondering the value?

Reply:
This is a very good question. There are three such methods `evaluate_gradient`: in `LinearExpression`, `QuadraticExpression`, and `QuadraticObjective`. The latter calls both from the expressions. All of them share the same signature. Yes, there's no need for the parameter in the `LinearExpression` implementation except for consistency. We may also come up with a base class for the expressions, not sure this will bring more clarity. So, I'd prefer to keep the method as it is now.

Reply:
Can I ask you add a reno yaml file - the changes are being managed via this tool now. There is information on how to install in contributing file https://github.com/Qiskit/qiskit-aqua/blob/master/CONTRIBUTING.md#release-notes